projects/meta_cl/experiments/anml_replicate.py

```diff
@@ -0,0 +1,202 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2020, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+import os
+from copy import deepcopy
+
+import numpy as np
+from PIL import Image
+from torch.optim import Adam
+from torchvision.transforms import Normalize, ToTensor
+
+from networks import ANMLNetwork
+
+from .oml import metacl_oml
+
+
+"""
+These experiments serve to narrow the gap between our implementation
+of ANML and the original repos.
+"""
+
+
+class ANMLTransform:
+    """
+    Transform images to RGB of size 28 x 28 and normalize with mean=0.92206 * 256
+    and std=0.08426 * 256.
+    """
+    def __init__(self):
+        self.to_tensor = ToTensor()
+        self.normalize = Normalize(mean=[0.92206 * 256] * 3,
+                                   std=[0.08426 * 256 * 256] * 3)
+
+    def __call__(self, pil_image):
+        img = pil_image.convert("L").convert("RGB")
+        img = img.resize((28, 28), resample=Image.LANCZOS)
+        img = np.array(img, dtype=np.float32)
+        img = self.to_tensor(img)
+        return self.normalize(img)
+
+
+# ANML-repo results:
+# |-----------------------------------------------------------|
+# | Num Classes | Meta test-test   | Meta test-train  |  LR   |
```

Comment:
This table is empty, should it be here?

Reply:
Good point, fix it. I meant to fill in the table after it was finished running.